app/src/lib/stores/pull-request-store.ts

```diff
@@ -139,9 +142,16 @@ export class PullRequestStore {
     api: API,
     repository: GitHubRepository
   ) {
-    const { name, owner } = getNameWithOwner(repository)
-    const open = await api.fetchAllOpenPullRequests(owner, name)
-    await this.storePullRequestsAndEmitUpdate(open, repository)
+    const bothPrLists = await Promise.all([
+      fetchOpenPullRequests(api, repository),
+      repository.fork && repository.parent
+        ? fetchOpenPullRequests(api, repository)
+        : Promise.resolve([]),
+    ])
+    await this.storePullRequestsAndEmitUpdate(
```

Comment:
`storePullRequestsAndEmitUpdate` was written with the intention that the PRs in the list would all target the same repository as the `repository` parameter is referring to and with this change that's no longer true. I'm finding it difficult to fully foresee the implications of this but it's making me nervous.

I think, for one thing, that it could break the delta updating algorithm since that relies on accurately keeping track of the most recently updated PR for a particular repository (that timestamp is then used in subsequent fetches). With this change I believe we could end up in a scenario where we'd be missing new or updated PRs from one of the repositories since we'd be storing the max of `updated_at` from either repository in the `pullRequestsLastUpdated` table for the fork repository.

I also think it will break our optimization to avoid updating the PR list when we get exactly one PR back: https://github.com/desktop/desktop/blob/2870bfb005e569d37c142d5c17f4f7138a44a20c/app/src/lib/stores/pull-request-store.ts#L380-L387

First of all there's always gonna be a minimum of two PRs in that list now (one for the fork and one for the parent) and even if that wasn't the case the `getPullRequest` call relies on the `repository` parameter to look up PRs targeting that repository and with this change it's not gonna find anything for the parent repository.

The last thing I thought of which is a minor thing is that I believe we could now be emitting an update event as a result of `storePullRequestsAndEmitUpdate` even if nothing has changed since `storePullRequests` could find a PR that has been updated for the parent that won't be returned in the `getAll` call.

Reply:
hey @niik i took a run at fixing this up, let me know what you think!